tensorflow/lite/kernels/internal/reference/non_max_suppression.h

```diff
@@ -18,111 +18,241 @@ limitations under the License.
 #include <algorithm>
 #include <cmath>
 #include <deque>
+#include <limits>
 #include <queue>
+#include <type_traits>
+
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/cppmath.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/types.h"
 
 namespace tflite {
 namespace reference_ops {
+namespace {
```

Comment:
Please don't use internal namespaces in header files.

https://google.github.io/styleguide/cppguide.html#Internal_Linkage

Should this and the implementations be moved to a `.cc` file instead?

Reply:
I renamed the namespace to `namespace detail` as done in other parts of TFLite.

If preferred, I can check to put the implementation in a `.cc` file and instantiate the template there.